### ISSUE TITLE
Path.normalize(): wrong interpretation of double forward slashes

### ISSUE THREAD

**Issue by MoritzBrueckner** [HaxeFoundation/haxe#10786]
Double forward slashes in paths seem to be interpreted as directories with empty names in `haxe.io.Path.normalize()`:

```haxe
class Test {
  static function main() {
    trace("Have:", haxe.io.Path.normalize("ProjectDir/Sources//../Assets"));
    // Have: ProjectDir/Sources/Assets
    trace("Want:", "ProjectDir/Assets");
  }
}
```
[Try it online](https://try.haxe.org/#F8AC8d8e)

The double forward slashes should be reduced to a single one and only then `../` should be evaluated. I'm a bit unsure whether the current situation is by design, but Python's `os.path.normpath()` for example works like intended with this example (have == want).

**Comment by haxiomic** [comment]
I agree with this, double forward slashes should be evaluated to a single slash first. Looks easy to fix, tho I wonder about existing code relying on this incorrect behavior!

**Comment by Aurel300** [comment]
Three or more slashes should also behave as a single one.
